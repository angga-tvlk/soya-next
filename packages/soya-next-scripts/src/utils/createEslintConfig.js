export default ({
  rules,
  ...eslintConfig
} = {
  rules: {},
}) => ({
  extends: 'marlint',
  globals: {
    soya: true,
  },
  rules: {
    'react/react-in-jsx-scope': 'off',
    ...rules,
  },
  ...eslintConfig,
});
