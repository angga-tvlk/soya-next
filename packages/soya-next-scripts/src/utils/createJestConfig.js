export default ({
  testMatch = ["**/__tests__/**/*.js?(x)", "**/?(*).(spec|test).js?(x)"],
  ...jestConfig
} = {}) => ({
  testMatch,
  ...jestConfig
});
