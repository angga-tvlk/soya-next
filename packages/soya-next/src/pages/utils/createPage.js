import createPageFactory from './createPageFactory';
import configureStore from '../../redux/configureStore';

export default createPageFactory(configureStore);
