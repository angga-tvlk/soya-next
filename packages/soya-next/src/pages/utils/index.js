import createPageFactory from './createPageFactory';

export { createPageFactory };
