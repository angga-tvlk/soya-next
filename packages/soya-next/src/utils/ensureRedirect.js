export default ({ method = 'GET', status = 301, ...redirect }) => ({
  method,
  status,
  ...redirect,
});
