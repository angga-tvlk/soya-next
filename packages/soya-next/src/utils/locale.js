import { parse as parseUrl } from 'url';

export const toPath = (locale, defaultLocale) => {
  if (locale && defaultLocale) {
    const localeArr = [];
    const [defaultLanguage, defaultCountry] = defaultLocale.split('-');
    const { country, language } = locale;
    if (language !== defaultLanguage) {
      localeArr.push(language);
    }
    if (country !== defaultCountry) {
      localeArr.push(country);
    }
    if (localeArr.length > 0) {
      return `/${localeArr.join('-')}`;
    }
  }
  return '';
};

export const ensurePath = (url, locale, defaultLocale) => {
  const localePrefix = toPath(locale, defaultLocale);
  if (localePrefix === '') {
    return url;
  }
  const { pathname } = parseUrl(url);
  const [localeSegment] = pathname.substr(1).split('/');
  if (localeSegment) {
    const [, defaultCountry] = defaultLocale.split('-');
    const [language, country = defaultCountry] = localeSegment.split('-');
    if (language === locale.language && country === locale.country) {
      return url;
    }
  }
  return localePrefix + url;
};

export const trimPath = (url, defaultLocale, siteLocales) => {
  if (defaultLocale && siteLocales) {
    const { pathname } = parseUrl(url);
    const [localeSegment] = pathname.substr(1).split('/');
    if (localeSegment) {
      const [, defaultCountry] = defaultLocale.split('-');
      const [language, country = defaultCountry] = localeSegment.split('-');
      if (siteLocales.indexOf(`${language}-${country}`) !== -1) {
        return url.substr(localeSegment.length + 1);
      }
    }
  }
  return url;
};
