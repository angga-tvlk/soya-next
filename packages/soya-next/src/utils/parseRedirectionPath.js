import pathToRegexp from "path-to-regexp";

export default (path, params) => pathToRegexp.compile(path)(params);
