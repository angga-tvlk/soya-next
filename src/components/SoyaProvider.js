import React from 'react';
import PropTypes from 'prop-types';
import { Cookies, CookiesProvider } from 'react-cookie';
import { Provider } from 'react-redux';
import { localeShape, urlShape } from '../constants/PropTypes';
import { storeShape } from 'react-redux/lib/utils/PropTypes';

class SoyaProvider extends React.Component {
  static propTypes = {
    store: storeShape.isRequired,
    cookies: PropTypes.instanceOf(Cookies).isRequired,
    defaultLocale: PropTypes.string,
    siteLocales: PropTypes.arrayOf(PropTypes.string.isRequired),
    locale: localeShape,
    url: urlShape,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = {
    defaultLocale: PropTypes.string,
    siteLocales: PropTypes.arrayOf(PropTypes.string.isRequired),
    locale: localeShape,
    url: urlShape,
  };

  getChildContext() {
    return {
      defaultLocale: this.props.defaultLocale,
      siteLocales: this.props.siteLocales,
      locale: this.props.locale,
      url: this.props.url,
    };
  }

  render() {
    return (
      <CookiesProvider cookies={this.props.cookies}>
        <Provider store={this.props.store}>
          {React.Children.only(this.props.children)}
        </Provider>
      </CookiesProvider>
    );
  }
}

export default SoyaProvider;
