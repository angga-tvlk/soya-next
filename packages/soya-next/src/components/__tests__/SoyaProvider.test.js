import React from 'react';
import PropTypes from 'prop-types';
import TestUtils from 'react-dom/test-utils';
import { Cookies } from 'react-cookie';
import { storeShape } from 'react-redux/lib/utils/PropTypes';
import SoyaProvider from '../SoyaProvider';
import { localeShape } from '../../constants/PropTypes';

describe('<SoyaProvider />', () => {
  const createMockStore = () => ({
    dispatch: () => jest.fn(),
    getState: () => jest.fn(),
    subscribe: () => jest.fn(),
  });

  const createChild = () => class extends React.Component {
    static contextTypes = {
      store: storeShape.isRequired,
      cookies: PropTypes.instanceOf(Cookies).isRequired,
      defaultLocale: PropTypes.string,
      siteLocales: PropTypes.arrayOf(PropTypes.string.isRequired),
      locale: localeShape,
    };

    render() {
      return <div />;
    }
  };

  it('should add cookies, defaultLocale, locale, siteLocales, and store to child context', () => {
    const Child = createChild();
    const tree = TestUtils.renderIntoDocument(
      <SoyaProvider
        cookies={new Cookies()}
        defaultLocale='id-id'
        locale={{ language: 'en', country: 'id' }}
        siteLocales={['id-id', 'en-id']}
        store={createMockStore()}
      >
        <Child />
      </SoyaProvider>
    );
    const child = TestUtils.findRenderedComponentWithType(tree, Child);
    expect(child.context).toMatchSnapshot();
  });
});
