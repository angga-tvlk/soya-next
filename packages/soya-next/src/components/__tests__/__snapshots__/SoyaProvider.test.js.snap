// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoyaProvider /> should add cookies, defaultLocale, locale, siteLocales, and store to child context 1`] = `
Object {
  "cookies": Cookies {},
  "defaultLocale": "id-id",
  "locale": Object {
    "country": "id",
    "language": "en",
  },
  "siteLocales": Array [
    "id-id",
    "en-id",
  ],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "subscribe": [Function],
  },
}
`;
